GeneratorInterface/Pythia8Interface/interface/Py8InterfaceBase.h

```diff
@@ -40,8 +40,8 @@ namespace gen {
     virtual void statistics();
     virtual const char* classname() const = 0;
 
-    void p8SetRandomEngine(CLHEP::HepRandomEngine* v) { p8RndmEngine_.setRandomEngine(v); }
-    P8RndmEngine& randomEngine() { return p8RndmEngine_; }
+    void p8SetRandomEngine(CLHEP::HepRandomEngine* v) { p8RndmEngine_->setRandomEngine(v); }
+    std::shared_ptr<P8RndmEngine> randomEngine() { return p8RndmEngine_; }
```

Comment:
Also this function could be kept to return `P8RndmEngine&`, as far as I can tell, 
```suggestion
    P8RndmEngine& randomEngine() { return *p8RndmEngine_; }
```